README
This archive contains the distribution Alien-UDUNITS2,
version 0.006_01:

  Alien package for the UDUNITS-2 physical unit manipulation and conversion library

This software is copyright (c) 2014 by Zakariyya Mughal.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v6.012.


Changes
0.008 «»
  - Update download URL. See
    <https://github.com/EntropyOrg/p5-Alien-UDUNITS2/pull/14>.
  - Extract cflags/libs for share install to alienfile. See
    <https://github.com/EntropyOrg/p5-Alien-UDUNITS2/pull/15>.

0.007 2020-05-10 16:39:02-0500
  - No changes since 0.006_01.

0.006_01 2020-05-07 16:10:51-0500
  - Use HTTPS for download. Thanks to thibaultduponchelle!
    See <https://github.com/EntropyOrg/p5-Alien-UDUNITS2/pull/10>,
    <https://github.com/EntropyOrg/p5-Alien-UDUNITS2/pull/11>.

0.006 2016-04-17 13:27:17-0500
  - fix previous release; changes are the same

0.005 2016-04-17 13:13:27-0500
  - Added a minimum version version for Alien::Base `configure_requires`.
    Thanks to Graham Ollis (PLICEASE++)!

0.004 2015-01-24 14:38:31-0600
  - add ExtUtils::Depends support

0.003 2014-09-14 21:38:28-0500
  - fix testing: floating point equality
 
0.002 2014-09-14 17:31:48-0500 
  - make sure Inline::C test is run

0.001 2014-09-14 17:09:43-0500
  - initial release


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.012.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Alien package for the UDUNITS-2 physical unit manipulation and conversion library",
  "AUTHOR" => "Zakariyya Mughal <zmughal\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Alien::Build" => "0.32",
    "Alien::Build::MM" => "0.32",
    "ExtUtils::MakeMaker" => "6.52"
  },
  "CONFIGURE_REQUIRES" => {
    "Alien::Build" => "0.32",
    "Alien::Build::MM" => "0.32",
    "ExtUtils::MakeMaker" => "6.52"
  },
  "DISTNAME" => "Alien-UDUNITS2",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Alien::UDUNITS2",
  "PREREQ_PM" => {
    "Alien::Base" => "0.038",
    "Exporter" => 0,
    "File::Spec" => 0,
    "Perl::OSType" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "File::Basename" => 0,
    "Module::Load" => 0,
    "Test::More" => 0,
    "Test::Number::Delta" => 0
  },
  "VERSION" => "0.006_01",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Alien::Base" => "0.038",
  "Alien::Build" => "0.32",
  "Alien::Build::MM" => "0.32",
  "Exporter" => 0,
  "ExtUtils::MakeMaker" => "6.52",
  "File::Basename" => 0,
  "File::Spec" => 0,
  "Module::Load" => 0,
  "Perl::OSType" => 0,
  "Test::More" => 0,
  "Test::Number::Delta" => 0,
  "strict" => 0,
  "warnings" => 0
);


# BEGIN code inserted by Dist::Zilla::Plugin::AlienBuild
use Alien::Build::MM;
my $abmm = Alien::Build::MM->new;
%WriteMakefileArgs = $abmm->mm_args(%WriteMakefileArgs);
# END code inserted by Dist::Zilla::Plugin::AlienBuild


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


# BEGIN code inserted by Dist::Zilla::Plugin::AlienBuild
{ package
    MY;
  sub postamble {
    $abmm->mm_postamble(@_);
  }
  sub install {
    $abmm->can('mm_install')
      ? $abmm->mm_install(@_)
      : shift->SUPER::install(@_);
  }
}
# END code inserted by Dist::Zilla::Plugin::AlienBuild



lib/Alien/UDUNITS2.pm
package Alien::UDUNITS2;

use strict;
use warnings;

require Alien::Base;
require Exporter;
our @ISA = qw(Alien::Base Exporter);
our @EXPORT_OK = qw(Inline);
use Perl::OSType qw(os_type);
use File::Spec;

sub inline_auto_include {
	[ 'udunits2.h' ];
}

sub Inline {
	my ($class, $lang) = @_;
	return unless $lang eq 'C'; # Inline's error message is good
	my $params = Alien::Base::Inline(@_);

	# Use static linking instead of dynamic linking. This works
	# better on some platforms. On macOS, to use dynamic linking,
	# the `install_name` of the library must be set, but since this
	# is the final path by default, linking to the `.dylib` under
	# `blib/` at test time does not work without using `@rpath`.
	if( $^O eq 'darwin' and $class->install_type eq 'share' ) {
		$params->{MYEXTLIB} .= ' ' .
			join( " ",
				map { File::Spec->catfile(
					File::Spec->rel2abs($class->dist_dir),
					'lib',  $_ ) }
				qw(libudunits2.a)
			);
		$params->{LIBS} =~ s/-ludunits2//g;
	}

	$params;
}

sub units_xml {
	my ($self) = @_;

	my ($file) = grep
		{ -f }
		map {
			(
				"$_/share/xml/udunits/udunits2.xml",
				"$_/share/udunits/udunits2.xml",
				"$_/lib/udunits2.xml"
			)
		} (
			$self->install_type eq 'share'
			? $self->dist_dir
			: "/usr"
		);

	$file;
}

1;

__END__
# ABSTRACT: Alien package for the UDUNITS-2 physical unit manipulation and conversion library

=pod

=head1 Inline support

This module supports L<Inline's with functionality|Inline/"Playing 'with' Others">.

=head1 SEE ALSO

L<UDUNITS-2|http://www.unidata.ucar.edu/software/udunits/>


lib/Alien/UDUNITS2/Install/Files.pm
package Alien::UDUNITS2::Install::Files;
$Alien::UDUNITS2::Install::Files::VERSION = '0.005';
# allows other packages to use ExtUtils::Depends like so:
#   use ExtUtils::Depends;
#   my $p = new ExtUtils::Depends MyMod, Alien::UDUNITS2;
# and their code will have all UDUNITS2 available at C level

use strict;
use warnings;

use Alien::UDUNITS2 qw(Inline);
BEGIN { *Inline = *Alien::UDUNITS2::Inline }
sub deps { () }

__END__

=pod

=encoding UTF-8

=head1 NAME

Alien::UDUNITS2::Install::Files

=head1 VERSION

version 0.005

=head1 AUTHOR

Zakariyya Mughal <zmughal@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2014 by Zakariyya Mughal.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::Alien v0.023.
use strict;
use warnings;

use Module::Build 0.28;
use lib qw{inc}; use Alien::Base::ModuleBuild;

my %module_build_args = (
  "alien_build_commands" => [
    "%c --prefix=%s",
    "make"
  ],
  "alien_inline_auto_include" => [],
  "alien_name" => "UDUNITS2",
  "alien_repository" => {
    "host" => "ftp.unidata.ucar.edu",
    "location" => "/pub/udunits/",
    "pattern" => "^udunits-([\\d\\.]+)\\.tar\\.gz\$",
    "protocol" => "ftp"
  },
  "configure_requires" => {
    "Alien::Base::ModuleBuild" => "0.005",
    "File::ShareDir" => "1.03",
    "perl" => "5.006"
  },
  "dist_abstract" => "Alien package for the UDUNITS-2 physical unit manipulation and conversion library",
  "dist_author" => [
    "Zakariyya Mughal <zmughal\@cpan.org>"
  ],
  "dist_name" => "Alien-UDUNITS2",
  "dist_version" => "0.005",
  "license" => "perl",
  "module_name" => "Alien::UDUNITS2",
  "recursive_test_files" => 1,
  "requires" => {
    "Alien::Base" => "0.006",
    "Exporter" => 0,
    "File::ShareDir" => "1.03",
    "File::Spec" => 0,
    "Perl::OSType" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "test_requires" => {
    "File::Basename" => 0,
    "Inline" => "0.67",
    "Inline::C" => "0.62",
    "Module::Load" => 0,
    "Test::More" => 0,
    "Test::Number::Delta" => 0,
    "perl" => "5.006"
  }
);


my %fallback_build_requires = (
  "File::Basename" => 0,
  "Inline" => "0.67",
  "Inline::C" => "0.62",
  "Module::Load" => 0,
  "Test::More" => 0,
  "Test::Number::Delta" => 0,
  "perl" => "5.006"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Alien::Base::ModuleBuild->new(%module_build_args);


$build->create_build_script;
